js/phyx.js

```diff
@@ -368,8 +368,8 @@ class TaxonomicUnitWrapper {
       });
     }
 
-    // If we don't have any properties of a taxonomic unit, return undefined.
-    if (labels.length === 0) return undefined;
+    // If we don't have any properties of a taxonomic unit, return 'Empty taxonomic unit'.
+    if (labels.length === 0) return 'Empty taxonomic unit';
```

Comment:
Can you explain? This seems like a bad idea in multiple ways:
- It defines _'Empty taxonomic unit'_ by fiat in a comment as a string with very special meaning. Put differently, the comment doesn't really make sense anymore, because the part `return 'Empty taxonomic unit'` adds absolutely zero information on top of the following line of code and can just be removed, but without it the comment that remains is a half sentence left dangling.
- It hardcodes the special meaning string rather than using a constant defined elsewhere.

So how is returning this special valued string more consistent with what one should expect to be returned than `undefined`?

Reply:
You're right, and I made the same mistake in the getSpecifierLabel code. I've fixed both of them in f52356a -- the model (phyx.js) now returns undefined for (1) an "empty" taxonomic unit (i.e. one without a scientific name, specimen identifier or external link), (2) a specifier that contains only empty taxonomic units or (3) a specifier that contains no taxonomic units at all. However, the controller (curation-tool.js) will replace undefined labels with a text string (e.g. "Empty specifier") when displaying it in the view (index.html). I could have incorporated it into the view directly, but that would make the code there overly complicated. What do you think?

Reply:
It seems like how to display something is the responsibility of the view, and not the model or the controller?